uat/custom-components/client-mosquitto-c/src/MqttConnection.cpp

```diff
@@ -280,15 +284,15 @@ std::vector<int> MqttConnection::subscribe(unsigned timeout, const int * subscri
     PendingRequest * request = 0;
     mosquitto_property * properties = NULL;
     int message_id = 0;
+    std::unique_ptr<char*> filters_array(new char*[filters.size()]);
```

Comment:
can we use std::array?

Reply:
But I found more stylish solution: just pass vector<string> instead of list<string> and use directly.

Reply:
I don't like the idea of allocating a raw array and I would prefer to use std array or vector if that is possible.

Reply:
Tried std::array  - the second parameter of template should be constant expression which is not true in our case, so std::array is not applicable here.

Tried to pass std::vector<std::string> directly but that is not possible due to std::string is a complex object but both mosquitto_unsubscribe_multiple() and mosquitto_subscribe_multiple() require  `pointer to array of pointers which points to ASCIIZ strings`.
Replace std::list<std::string> to std::vector<std::string> to speedup the code in e16d86dd49ff9ca4f5c8963b19bf86810da3d6a3